dockerdata/dnsdist-resolver.lua

```diff
@@ -0,0 +1,113 @@
+-- testing oneliner:
+-- resolver = require 'dnsdist-resolver' resolver.maintenance() resolver.servers['www.7bits.nl']={pool='blabla'} resolver.maintenance() os.execute('sleep 3') resolver.maintenance() showServers() resolver.servers['www.7bits.nl']=nil resolver.maintenance() os.execute('sleep 3') resolver.maintenance() showServers()
+
+local _M = {}
+
+-- these are the servers we want - somebody should populate it
+-- example:
+--  resolver.servers['ns.example.com'] = { pool='auths', order=3 }
+-- do not set name, address, id
+_M.servers = {}
+
+-- these are the servers we have
+-- key = name
+-- value = {address, serverObject} (should make these named members)
+local ourservers = {}
+
+local resolverpipe = io.popen('/usr/local/bin/dnsdist-resolver', 'w')
+
+local function tablecopy(t)
+    local t2 = {}
+    for k,v in pairs(t)
+    do
+        t2[k] = v
+    end
+    return t2
+end
+
+local function removeServer(name)
+    rmServer(ourservers[name][2])
+    ourservers[name] = nil
+end
+
+local function setServer(name, ip)
+    -- adds a server or changes its IP
+    local existing = ourservers[name]
+    if existing ~= nil
+    then
+        -- it exists, check IP
+        infolog(string.format("existing[1] [%s] == ip [%s] ??", existing[1], ip))
+        if existing[1] == ip
+        then
+            -- IP is correct, done!
+            return
+        else
+            -- IP is wrong, drop and re-add it
+            removeServer(name)
+        end
+    end
+
+    -- it does not exist, let's add it
+    local settings = tablecopy(_M.servers[name])
+    settings.name = name
+    -- TODO: we only take the first IP
+    settings.address = ip
+    ourservers[name] = {ip, newServer(settings)}
+end
+
+function _M.maintenance()
+    -- TODO: only do this if the list has changed
+    -- TODO: check return values
+    for k,v in pairs(_M.servers)
+    do
+        resolverpipe:write(k..' ')
+    end
+    resolverpipe:write('\n')
+    resolverpipe:flush()
+
+    -- TODO: maybe this failure should be quiet for the first X seconds?
+    local ret, resout = pcall(loadfile, './dnsdist-resolver.out')
```

Comment:
This looked like a race condition in that you cannot be sure it has already been written. It does not matter, because it will be read during a future maintenance. Perhaps add a comment explaining this intended behavior?